services/trainer/trainer/trainer.py

```diff
@@ -0,0 +1,90 @@
+from dataclasses import dataclass
```

Comment:
Triggered! This reminds me of `/elpis/elpis/` 😅

Reply:
This is because of the nested poetry structure:
- The outer trainer is the entire folder for the service
- The inner trainer fikder is the python package name
- The trainer file is where the training logic takes place within the package
It does look gross though- what do you recommend?

Reply:
I'd probably aim for something like `services/trainer/src/trainer.py`. Because of the way poetry works you should be able reference the module elsewhere with `import trainer.src.trainer as trainer; trainer.blah()` outside the trainer file if you need to.